api/src/main/java/org/apache/iceberg/IncrementalScan.java

```diff
@@ -33,6 +49,21 @@
    */
   ThisT fromSnapshotInclusive(long fromSnapshotId);
 
+  /**
+   * Instructs this scan to look for changes starting from a particular snapshot (exclusive).
+   *
+   * <p>If the start snapshot is not configured, it is defaulted to the oldest ancestor of the end
+   * snapshot (inclusive).
+   *
+   * @param fromSnapshotId the start snapshot ID (exclusive)
+   * @param branch the ref used
+   * @return this for method chaining
+   * @throws IllegalArgumentException if the start snapshot is not an ancestor of the end snapshot
+   */
+  default ThisT fromSnapshotExclusive(long fromSnapshotId, String branch) {
```

Comment:
I think these APIs needs to enforce that fromSnapshotId is an ancestor of the branch.

Reply:
Done